### PR TITLE
Update to newer version of Snes9x

### PR DESCRIPTION
**Pull request by Citrodata**
Update to newer version of Snes9x, while also updating most of the dependencies and the platform to 23.08

### CODE REVIEW

**Comment by Citrodata** [comment]
@bbhtt Anything missing or can this be merged to finnally update this like nestopia?

File: com.snes9x.Snes9x.json [hunk 1]
```diff
@@ -2,7 +2,7 @@
     "app-id": "com.snes9x.Snes9x",
     "runtime": "org.freedesktop.Platform",
     "sdk": "org.freedesktop.Sdk",
-    "runtime-version": "22.08",
+    "runtime-version": "23.08",
```
Comment (bbhtt) [inline com.snes9x.Snes9x.json:5]: can you bump to 24.08?

**Comment by bbhtt** [comment]
can you fix the commit messages to have some description of the change? `Update foobar to v1.0.0`

**Comment by Citrodata** [comment]
How can i edit the old commits?

File: com.snes9x.Snes9x.json (hunk 1 above)
Comment (Citrodata) [reply com.snes9x.Snes9x.json]: Updated it

**Comment by bbhtt** [comment]
> How can i edit the old commits?

`git rebase -i HEAD~10` then `change `pick` to `reword` for the commits you want to edit.

**Comment by Citrodata** [comment]
I sadly currently have no pc with git.
Should i just create a new RP with one commit message?

**Comment by bbhtt** [comment]
It is not possible to do without git. You should install it.

**Comment by Citrodata** [comment]
Okay. Cause i only update everything via github ui.

**Comment by Citrodata** [comment]
@bbhtt I think this could get merged now. For the Future i will document the dependencie changes.

**Comment by bbhtt** [comment]
I will edit the commits tomorrow and merge it but in the future please use `git` and proper commit messages.

File: com.snes9x.Snes9x.json [hunk 2]
```diff
@@ -37,8 +37,21 @@
...
+        {
+            "name": "sigc++",
+            "config-opts": [
+                "--disable-documentation"
+            ],
+            "sources": [
+                {
+                    "type": "archive",
+                    "url": "https://download.gnome.org/sources/libsigc%2B%2B/3.6/libsigc%2B%2B-3.6.0.tar.xz",
+                    "sha256": "c3d23b37dfd6e39f2e09f091b77b1541fbfa17c4f0b6bf5c89baef7229080e17"
```
Comment (bbhtt) [inline com.snes9x.Snes9x.json:54]: you don't need both sigc++ versions?

File: com.snes9x.Snes9x.json [hunk 3]
```diff
@@ -50,8 +63,8 @@
             "sources": [
                 {
                     "type": "archive",
-                    "url": "https://cairographics.org/releases/cairomm-1.14.3.tar.xz",
-                    "sha256": "0d37e067c5c4ca7808b7ceddabfe1932c5bd2a750ad64fb321e1213536297e78"
+                    "url": "https://www.cairographics.org/releases/cairomm-1.14.5.tar.xz",
```
Comment (bbhtt) [inline com.snes9x.Snes9x.json:66]:
```suggestion
                    "url": "https://cairographics.org/releases/cairomm-1.14.5.tar.xz",
```

File: com.snes9x.Snes9x.json (hunk 2 above)
Comment (Citrodata) [reply com.snes9x.Snes9x.json]: Checked it and its not needed. Removed it.

**Comment by Citrodata** [comment]
@bbhtt Still compiles with the changes. Also wrote the commit message to remove the unneeded dependency hopefully better this time.
And also sorry for making this merge harder than it should have been.

**Pull request merged**